regress/pesterTests/FileBasedLogging.tests.ps1

```diff
@@ -0,0 +1,237 @@
+                    ssh-keygen.exe -t ed25519 -f $KeyFilePath -P `"`"
+                }
+
+                Copy-Item "$keyFilePath.pub" $authorizedkeyPath -Force -ErrorAction SilentlyContinue
+
+                Repair-AuthorizedKeyPermission -Filepath $authorizedkeyPath -confirm:$false
+            }
+
+            $AdminUserProfile = $OpenSSHTestInfo['AdminUserProfile']
+            $NonAdminUserProfile = $OpenSSHTestInfo['NonAdminUserProfile']
+
+            $KeyFileName = $nonadminusername + "_sshtest_fileBasedLog_ed25519"
+            $NonadminKeyFilePath = Join-Path $testDir $keyFileName
+            Remove-Item -path "$NonadminKeyFilePath*" -Force -ErrorAction SilentlyContinue
+            Setup-KeyBasedAuth -Username $nonadminusername -KeyFilePath $NonadminKeyFilePath -UserProfile $NonAdminUserProfile
+
+            $KeyFileName = $adminusername + "_sshtest_fileBasedLog_ed25519"
+            $AdminKeyFilePath = Join-Path $testDir $keyFileName
+            Remove-Item -path "$AdminKeyFilePath*" -Force -ErrorAction SilentlyContinue
+            Setup-KeyBasedAuth -Username $adminusername -KeyFilePath $AdminKeyFilePath -UserProfile $AdminUserProfile
+
+            #create batch file
+            $commands = 
+"ls
+exit"
+
+            $batchFilePath = Join-Path $testDir "$tC.$tI.commands.txt"
+            Set-Content $batchFilePath -Encoding UTF8 -value $commands
+
+            # clear logs so that next testcase will get fresh logs.
+            Clear-Content "$env:ProgramData\ssh\logs\sftp-server.log" -Force -ErrorAction SilentlyContinue
+
+            $tI = 1
+		}
+		
+        BeforeEach {
+            #clean sftp log file
+            $sshlog = Join-Path $testDir "$tC.$tI.$sshLogName"            
+            $sshdlog = Join-Path $testDir "$tC.$tI.$sshdLogName"
+            if(-not $skip)
+            {
+                Stop-SSHDTestDaemon   -Port $port
+            }
+        }
+
+        AfterAll {
+            Remove-Item -path "$NonadminKeyFilePath*" -Force -ErrorAction SilentlyContinue
+            Remove-Item -path "$AdminKeyFilePath*" -Force -ErrorAction SilentlyContinue
+
+            $tC++
+        }
+
+        It "$tC.$tI-Nonadmin SFTP Connection"  -skip:$skip {
+
+            Start-SSHDTestDaemon -WorkDir $opensshbinpath -Arguments "-ddd -f $sshdConfigPath -E $sshdlog" -Port $port
+
+            sftp -P $port -i $NonadminKeyFilePath -b $batchFilePath $nonadminusername@$server
+
+            Stop-SSHDTestDaemon   -Port $port
+
+            #Copy sftp-log files into test directory
+            $sftplog = Join-Path $testDir "$tC.$tI.sftp-server.log"
+            Copy-Item "$env:ProgramData\ssh\logs\sftp-server.log" $sftplog -Force -ErrorAction SilentlyContinue
+   
+            # clear logs so that next testcase will get fresh logs.
+            Clear-Content "$env:ProgramData\ssh\logs\sftp-server.log" -Force -ErrorAction SilentlyContinue
+
+            $sshdlog | Should Contain "Accepted publickey for $nonadminusername"
```

Comment:
global comment. Add "sleep 3" after Stop-SSHDTestDaemon. Please fix this in all existing tests as well